### PR TITLE
host delete permissions test

### PR DESCRIPTION
**Pull request by pnovotny**
SAT-21619 (component evaluation).

Check for correct user permissions when deleting a host.

### CODE REVIEW

**Comment by pnovotny** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_permission.py -k test_positive_check_delete

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5879
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_permission.py -k test_positive_check_delete --external-logging
Test Result : ===== 1 failed, 3 passed, 12 deselected, 68 warnings in 712.09s (0:11:52) ======
```

**Comment by pnovotny** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_permission.py -k TestUserRole

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5923
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_permission.py -k TestUserRole --external-logging
Test Result : ========== 13 passed, 3 deselected, 186 warnings in 850.49s (0:14:10) ==========
```

**Review by Griffin-Sullivan — APPROVED** [review]

File: tests/foreman/api/test_permission.py [hunk 1]
```diff
@@ -161,18 +161,20 @@ def test_positive_search(self, target_sat):
 
 # FIXME: This method is a hack. This information should somehow be tied
 # directly to the `Entity` classes.
-def _permission_name(entity, which_perm):
-    """Find a permission name.
+def _permission_names(entity, which_perm: str) -> list:
+    """Find a permission names.
```
Comment (lhellebr) [inline tests/foreman/api/test_permission.py:165]:
```suggestion
    """Find permission names.
```

**Review by lhellebr — APPROVED** [review]

File: tests/foreman/api/test_permission.py (hunk 1 above)
Comment (pnovotny) [reply tests/foreman/api/test_permission.py]: Fixed

**Comment by pnovotny** [comment]
The Code Quality failure is a known issue and it's addressed here: https://github.com/JacobCallahan/ssh2-python/pull/1

**Review by synkd — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

**Pull request merged**